salesagility/SuiteCRM
# Fix #9676 - Control the system-generated password process

Solves #9676 

## Description
A validation is added when saving the configuration for the process of managing passwords, so that if the _Enable System-Generated Passwords Feature_ property is activated, it is verified that there is a template indicated in the _For system-generated password_ field

In addition, in the process of creating a new user, the value of the _Enable System Generated Passwords Feature_ property is checked so that if it is activated, the email will be sent to the user and if it is not activated, it will not.

## Motivation and Context
Resolve conflicts in the system-generated password process

## How To Test This
1. Go to Password Management
2. Click the _Enable System Generated Passwords Feature_ option and select the None option in the field _For system generated password_
3. Click on Save and check that a validation error is displayed in the field _For system generated password_
4. Uncheck the _Enable System Generated Passwords Feature_ option and select the System Generated Password template in the field _For system generated password_
5. Create a new user and check that no email arrives indicating a password generated automatically by the system.

## Types of changes


### Final checklist
